packages/babel-traverse/src/path/family.js

```diff
@@ -67,6 +67,39 @@ export function getSibling(key) {
   });
 }
 
+export function getPrevSibling() {
+  return this.getSibling(this.key - 1);
+}
+
+export function getNextSibling() {
+  return this.getSibling(this.key + 1);
+}
+
+export function getAllNextSibling(callback, thisArg = this) {
+  let _key = this.key;
+  let sibling = this.getSibling(++_key);
+  let siblings = [];
+  const hasCallback = callback && typeof callback === "function";
+  while (sibling.node) {
+    siblings = siblings.concat(hasCallback ? callback.call(thisArg, sibling) : sibling);
+    sibling = this.getSibling(++_key);
+  }
+  return siblings;
+}
+
+export function getAllPrevSibling(callback, thisArg = this) {
```

Comment:
I'm not a fan of this kind of API, where we mix callback and return. `getAllPrevSibling` sounds weird (vs. `getAllPrevSiblings`). Why not just `getAllPrevSiblings` to return all siblings, which could then be iterated over with `getAllPrevSiblings.forEach(sibling => ...)`